HKMP/Game/Client/PlayerManager.cs

```diff
@@ -69,19 +111,77 @@ internal class PlayerManager {
+
+            // Disable non bouncer component
+            var nonBouncer = playerPrefab.GetComponent<NonBouncer>();
+            nonBouncer.active = false;
+
             // Add some extra gameObjects related to animation effects
-            new GameObject("Attacks") {layer = 9}.transform.SetParent(_playerPrefab.transform);
-            new GameObject("Effects") {layer = 9}.transform.SetParent(_playerPrefab.transform);
-            new GameObject("Spells") {layer = 9}.transform.SetParent(_playerPrefab.transform);
+            new GameObject("Attacks") { layer = 9 }.transform.SetParent(playerPrefab.transform);
+            new GameObject("Effects") { layer = 9 }.transform.SetParent(playerPrefab.transform);
+            new GameObject("Spells") { layer = 9 }.transform.SetParent(playerPrefab.transform);
 
-            _playerPrefab.SetActive(false);
-            Object.DontDestroyOnLoad(_playerPrefab);
+            playerPrefab.transform.SetParent(_playerContainerPrefab.transform);
 
-            // Register packet handlers
-            packetManager.RegisterClientPacketHandler<ClientPlayerTeamUpdate>(ClientPacketId.PlayerTeamUpdate,
-                OnPlayerTeamUpdate);
-            packetManager.RegisterClientPacketHandler<ClientPlayerSkinUpdate>(ClientPacketId.PlayerSkinUpdate,
-                OnPlayerSkinUpdate);
+            CreateUsername(_playerContainerPrefab);
+
+            _playerContainerPrefab.SetActive(false);
+            Object.DontDestroyOnLoad(_playerContainerPrefab);
+
+            const int numThreads = 4;
+            const int loopsPerThread = InitialPoolSize / numThreads;
+
+            var threads = new List<Thread>(numThreads);
+
+            var syncLock = new object();
+
+            // Split instantiation of player pool into threads for speed
+            for (ushort threadNum = 0; threadNum < numThreads; threadNum++) {
+                var thread = new Thread(() => {
+                    for (ushort loopNum = 0; loopNum < loopsPerThread; loopNum++) {
+                        var playerContainer = Object.Instantiate(_playerContainerPrefab);
+                        Object.DontDestroyOnLoad(playerContainer);
+                        playerContainer.name = "Player Container";
+
+                        // Prevent other threads from modifying the queue at the same time as another to avoid undefined behavior
+                        lock (syncLock) {
+                            _inactivePlayers.Enqueue(playerContainer);
+                        }
+                    }
+                });
+
+                thread.Start();
+                threads.Add(thread);
+            }
+
+            // Wait for the threads to stop executing.
+            foreach (var thread in threads) {
+                thread.Join();
+            }
+
+            foreach (var playerContainer in _inactivePlayers) {
+                MakeUniqueSpriteAnimator(playerContainer.FindGameObjectInChildren("PlayerPrefab"));
```

Comment:
Why is this done here and not during the instantiation of the player container within the thread?

Reply:
Putting it in a thread causes an access violation when `CopyUtil.SmartCopyMaterial` is called and makes the game crash.